services/scabbard/libscabbard/src/store/scabbard_store/diesel/mod.rs

```diff
@@ -0,0 +1,1077 @@
+        service_id: &FullyQualifiedServiceId,
+        epoch: u64,
+    ) -> Result<i64, ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection)
+            .add_consensus_action(action, service_id, epoch)
+    }
+    /// Update an existing 2 phase commit action
+    fn update_consensus_action(
+        &self,
+        service_id: &FullyQualifiedServiceId,
+        epoch: u64,
+        action_id: i64,
+        executed_at: SystemTime,
+    ) -> Result<(), ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).update_consensus_action(
+            service_id,
+            epoch,
+            action_id,
+            executed_at,
+        )
+    }
+    /// List all coordinator actions for a given service_id and epoch
+    fn list_consensus_actions(
+        &self,
+        service_id: &FullyQualifiedServiceId,
+        epoch: u64,
+    ) -> Result<Vec<ScabbardConsensusAction>, ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).list_consensus_actions(service_id, epoch)
+    }
+    /// List ready services
+    fn list_ready_services(&self) -> Result<Vec<FullyQualifiedServiceId>, ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).list_ready_services()
+    }
+    /// Add a new scabbard service
+    fn add_service(&self, service: ScabbardService) -> Result<(), ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).add_service(service)
+    }
+    /// Add a new commit entry
+    fn add_commit_entry(&self, commit_entry: CommitEntry) -> Result<(), ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).add_commit_entry(commit_entry)
+    }
+    /// Get the commit entry for the specified service_id and epoch
+    fn get_last_commit_entry(
+        &self,
+        service_id: &FullyQualifiedServiceId,
+    ) -> Result<Option<CommitEntry>, ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).get_last_commit_entry(service_id)
+    }
+    /// Update an existing commit entry
+    fn update_commit_entry(&self, commit_entry: CommitEntry) -> Result<(), ScabbardStoreError> {
+        ScabbardStoreOperations::new(self.connection).update_commit_entry(commit_entry)
+    }
+}
+
+#[cfg(all(test, feature = "sqlite"))]
+pub mod tests {
+    use super::*;
+
+    use std::time::Duration;
+    use std::time::SystemTime;
+
+    use crate::migrations::run_sqlite_migrations;
+
+    use crate::store::scabbard_store::{
+        commit::{CommitEntryBuilder, ConsensusDecision},
+        context::{ContextBuilder, Participant},
+        service::{ScabbardServiceBuilder, ServiceStatus},
+        state::Scabbard2pcState,
+        two_phase::action::{ConsensusAction, ConsensusActionNotification},
+    };
+
+    use diesel::{
+        r2d2::{ConnectionManager, Pool},
+        sqlite::SqliteConnection,
+    };
+    use splinter::{service::FullyQualifiedServiceId, service::ServiceId};
+
+    #[test]
```

Comment:
All of the tests should have documentation

Reply:
added docs for all tests